src/client.ts

```diff
@@ -355,6 +364,74 @@ export class Client {
     }
   }
 
+  /**
+ * Take a screenshot of the page and save it to `filename` in `path` folder, with a `format` and `quality` (jpeg format only)
+ * If `selector` is passed in, it will take a screenshot of only that element
+ * and its children as opposed to the whole page.
+ *
+ * @param path - The path of where to save the screenshot to
+ * @param options - options
+ * @param options.filename - Name to be given to the screenshot. Optional
+ * @param options.selector - Screenshot the given selector instead of the full page. Optional
+ * @param options.format - The Screenshot format(and hence extension). Allowed values are "jpeg", "png", and "webp" - Optional
+ * @param options.quality - The image quality from 0 to 100, default 80. Applicable only if no format provided or format is "jpeg" - Optional
+ */
+  public async takeScreenshot(
+    path: string,
+    options?: {
+      selector?: string;
+      fileName?: string;
+      format?: "jpeg" | "png" | "webp";
+      quality?: number;
+    },
+  ): Promise<void> {
+    if (!existsSync(path)) {
+      await this.done();
+      throw new Error(`The provided folder path - ${path} doesn't exist`);
+    }
+    let ext = "jpeg";
+    let quality: number | undefined = 80;
+    // deno-lint-ignore ban-types
+    let clip: Object | undefined = undefined;
+
+    if (options?.format) {
+      ext = options.format;
+      quality = (ext == "jpeg")
+        ? ((options.quality && Math.abs(options.quality) <= 100)
+          ? Math.abs(options.quality)
+          : 80)
+        : undefined;
+    }
+
+    if (options?.selector) {
+      clip = await this.getViewport(options.selector);
+    }
+
+    const res = await this.sendWebSocketMessage(
+      "Page.captureScreenshot",
+      {
+        format: "jpeg",
+        quality: quality,
+        clip: clip,
+      },
+    ) as {
+      data: string;
+    };
+
+    //Writing the Obtained Base64 encoded string to image file
+    let fName = (options?.fileName || generateTimestamp()) + `.${ext}`;
```

Comment:
So im guessing 99% of the time, a user will do `screenshot(path, { filename: "home_page.png" })`, but because your code adds the extension, the resulting file will be `./Screenshots/home_page.png.png`

Could we also use const here? There's also no need for the line below, windows will work fine using / for relative paths

Reply:
Replacing all jpg jpeg png and webp with "" and using the provided extension xD